tests/snapshots/guppy__llvmir@quantum-classical-1.snap

```diff
@@ -16,59 +16,39 @@ source_filename = "hugr-qir"
 
 cond_exit_77:                                     ; preds = %cond_77_case_1, %cond_77_case_0
-  %"0.0" = phi %QUBIT* [ null, %cond_77_case_0 ], [ %3, %cond_77_case_1 ]
-  call void @__quantum__qis__phasedx__body(double 0x3FF921FB54442D18, double 0xBFF921FB54442D18, %QUBIT* %"0.0")
-  call void @__quantum__qis__rz__body(double 0x400921FB54442D18, %QUBIT* %"0.0")
-  %7 = call %RESULT* @__quantum__qis__m__body(%QUBIT* %"0.0")
-  %8 = call i1 @__quantum__qis__read_result__body(%RESULT* %7)
-  %9 = select i1 %8, i1 true, i1 false
-  %10 = call %QUBIT* @__quantum__rt__qubit_allocate()
-  %11 = insertvalue { i1, %QUBIT* } { i1 true, %QUBIT* poison }, %QUBIT* %10, 1
-  %12 = extractvalue { i1, %QUBIT* } %11, 0
-  %cond41 = icmp eq i1 %12, true
-  br i1 %cond41, label %cond_90_case_1, label %cond_90_case_0
+  call void @__quantum__qis__phasedx__body(double 0x3FF921FB54442D18, double 0xBFF921FB54442D18, %QUBIT* null)
+  call void @__quantum__qis__rz__body(double 0x400921FB54442D18, %QUBIT* null)
+  %1 = call %RESULT* @__quantum__qis__m__body(%QUBIT* null)
+  %2 = call i1 @__quantum__qis__read_result__body(%RESULT* %1)
+  br i1 true, label %cond_90_case_1, label %cond_90_case_0
 
 cond_90_case_1:                                   ; preds = %cond_exit_77
-  %13 = extractvalue { i1, %QUBIT* } %11, 1
   br label %cond_exit_90
 
 cond_90_case_0:                                   ; preds = %cond_exit_77
-  %14 = extractvalue { i32, i8* } { i32 1, i8* getelementptr inbounds ([37 x i8], [37 x i8]* @1, i32 0, i32 0) }, 0
-  %15 = extractvalue { i32, i8* } { i32 1, i8* getelementptr inbounds ([37 x i8], [37 x i8]* @1, i32 0, i32 0) }, 1
-  %16 = call i32 (i8*, ...) @printf(i8* getelementptr inbounds ([34 x i8], [34 x i8]* @prelude.panic_template.1, i32 0, i32 0), i32 %14, i8* %15)
+  %3 = call i32 (i8*, ...) @printf(i8* getelementptr inbounds ([34 x i8], [34 x i8]* @prelude.panic_template.1, i32 0, i32 0), i32 1, i8* getelementptr inbounds ([37 x i8], [37 x i8]* @1, i32 0, i32 0))
   call void @abort()
   br label %cond_exit_90
 
 cond_exit_90:                                     ; preds = %cond_90_case_1, %cond_90_case_0
-  %"018.0" = phi %QUBIT* [ null, %cond_90_case_0 ], [ %13, %cond_90_case_1 ]
-  call void @__quantum__qis__phasedx__body(double 0x3FF921FB54442D18, double 0xBFF921FB54442D18, %QUBIT* %"018.0")
-  call void @__quantum__qis__rz__body(double 0x400921FB54442D18, %QUBIT* %"018.0")
-  %17 = call %RESULT* @__quantum__qis__m__body(%QUBIT* %"018.0")
-  %18 = call i1 @__quantum__qis__read_result__body(%RESULT* %17)
-  %19 = select i1 %18, i1 true, i1 false
-  %20 = select i1 %9, i1 true, i1 false
-  %21 = select i1 %19, i1 true, i1 false
-  %22 = xor i1 %21, %20
-  %23 = select i1 %22, i1 true, i1 false
-  call void @__quantum__rt__bool_record_output(i1 %23, i8* getelementptr inbounds ([2 x i8], [2 x i8]* @2, i32 0, i32 0))
+  %"018.0.reg2mem.0" = phi %QUBIT* [ inttoptr (i64 1 to %QUBIT*), %cond_90_case_1 ], [ null, %cond_90_case_0 ]
```

Comment:
From my understanding we still have the panic and abort calls left which we need to remove. I would try in a next step to check all commands and remove them if they are prints / aborts?

Reply:
Sounds like a good idea, I will try this in a separate PR.